### DISCUSSION TITLE
Which scientific discoveries led to the creation of fingerprintjs?

### DISCUSSION THREAD

**Discussion by shenaseh** [fingerprintjs/fingerprintjs#843] [General]
I'm going to review a free* software for my university thesis, and the open source fingerprintjs looks appealing.
Is it documented how it works to generate a scored fingerprint id?
If its algorithm(s) are based on scientific articles, could you please provide those references? I would appreciate any advice in this matter.

---
\* Free as freedom, not a free beer!

**Comment by makma** [comment]
Hey @shenaseh,

> Is it documented how it works to generate a scored fingerprint id?
- All code is well commented.
- What do you mean by the `scored fingerprint`?

> If its algorithm(s) are based on scientific articles, could you please provide those references? I would appreciate any advice in this matter.

A good scientific content for start might be [EssentialFP: Exposing the Essence of Browser Fingerprinting](https://www.cse.chalmers.se/research/group/security/essential-fp/secweb-21.pdf)

**Reply by shenaseh** [reply]
> What do you mean by the scored fingerprint?

The degree of uniqueness of the identifier (fingerprint) generated from the user's device


> A good scientific content for start might be [EssentialFP: Exposing the Essence of Browser Fingerprinting](https://www.cse.chalmers.se/research/group/security/essential-fp/secweb-21.pdf)

❤️

> All code is well commented.

Yes, I am deciphering the code (:
Agreed. Well-commented. I just thought you might have some advice to assist me forward.

**Reply by makma** [reply]
> The degree of uniqueness of the identifier (fingerprint) generated from the user's device

The library does not provide or consider such a metric. It just generates a hash from the provided signals without grading.

**Reply by shenaseh** [reply]
If not, then what is meant by _Accuracy: up to 60%_ in the README page description?

**Reply by makma** [reply]
That's an estimation based on our internal data and might differ on the specific environment.
What we mean by the accuracy is explained in this article: https://fingerprint.com/blog/device-fingerprinting-accuracy/

**Comment by perpi** [comment]
> That's an estimation based on our internal data

Do you mean the formula below?
![image](https://user-images.githubusercontent.com/7093244/209407386-aa5f6b2e-9551-48ab-a38a-017c13bf7a30.png)

> might differ on the specific environment.

Would you explain more about how the environment has a role?

> The library does not provide or consider such a metric.

[Here](https://github.com/fingerprintjs/fingerprintjs/blob/master/src/confidence.ts),  in the open source version, you have something called the _Confidence_. By choosing this word, how do you differentiate it from the word "accuracy"? And, what purpose can it be used for?
Also, why is it measured so simply and in terms of **only one** general variable (the type of user's operating system)?
